core/frontend/src/utils/helper_functions.ts

```diff
@@ -0,0 +1,18 @@
+/* Asynchronously wait for a given interval. */
+/**
+ * @param interval - Time in milliseconds to wait after the previous request is done
+* */
+export function sleep(interval: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, interval))
+}
+
+/* Call a given function periodically for a given interval. */
+/**
+ * @param func - Function to be called.
+ * @param interval - Time in milliseconds to wait after the previous request is done
+* */
+export async function callPeriodically(func: () => Promise<void>, interval: number): Promise<void> {
+  await func()
+  await sleep(interval)
+  callPeriodically(func, interval)
+}
```

Comment:
That's a bit dangerous, how can we turn off the `callPeriodically` if necessary ?

Reply:
Let me give a double-check, but as it's a set-interval I think it will die for itself when the component is mounted.

Is this your question or your concerned about something else?

Reply:
I'm concerned about:
```
let potato = {
    a: async () => console.log("Potato")
}

callPeriodically(potato.a, 1000)

# somehow potato is dead
```

What happens if the object that callPeriodically is using dies or the variables used inside potato.a dies ?

Reply:
ping @Williangalvani

Reply:
Opened #425 to discuss that as it was already merged with #391